api/contacts.py

```diff
@@ -69,35 +77,48 @@ def setResponse(self, response: Response):
 
 
 class GetContactById(GenericTransaction):
-    '''
+    """
     Retrieve contacts with offset and limit
-    '''
+    """
+
     def __init__(self, id: int):
         self.request = Request(Endpoint.CONTACTS, Method.GET, {"id": id})
 
     def setResponse(self, response: Response):
-        self.response = Contacts(response)
+        self.response = ContactByIdEntry(response)
 
 
 class AddContact(GenericTransaction):
-    '''
+    """
     Retrieve contacts with offset and limit
-    '''
+    """
+
     def __init__(self, newContactEntry):
-        try:
-            self.request = Request(Endpoint.CONTACTS, Method.POST, newContactEntry)
-        except TransactionError as err:
-            pass
+        self.request = Request(Endpoint.CONTACTS, Method.POST, newContactEntry)
 
     def setResponse(self, response: Response):
         self.response = AddedContactId(response)
 
+
+class UpdateContact(GenericTransaction):
+    """
+    Retrieve contacts with offset and limit
+    """
+
+    def __init__(self, updatedContactRecord):
+        self.request = Request(Endpoint.CONTACTS, Method.PUT, updatedContactRecord)
+
+    def setResponse(self, response: Response):
+        self.response = UpdatedContactId(response)
+
+
 class DeleteContactById(GenericTransaction):
-    '''
+    """
     Delete contact with specified by Id
-    '''
+    """
+
     def __init__(self, id: int):
         self.request = Request(Endpoint.CONTACTS, Method.DEL, {"id": id})
 
     def setResponse(self, response: Response):
-        self.response = DeletedContact(response)
+        self.response = GenericResponse(response)
```

Comment:
Deleted contact added nothing, so we just return generic response? :) Do we need to override setResponse then? :)

Reply:
`setResponse` needs to be overridden.
Without it we have error:
`E       TypeError: Child class has to implement setResponse  `
Then we use status returned after deleting contact (it is in the response).

Reply:
tbh, I forgot how I wrote it. Makes sense then